### PR TITLE
fix(mediaplayerelement): fix MPE position when scrolling

### LINKED ISSUE #13954: [MediaPlayerElement][Skia] The media player element moves with the screen
### Current behavior

skiaweirdo
The media player element moves along the screen when you scroll. 

### Expected behavior

The mediaplayer should stay at it's place in the screen just like on all the other platforms.

### How to reproduce it (as minimally and precisely as possible)

```
<ScrollViewer>
    <StackPanel Spacing="20">
        <MediaPlayerElement x:Name="MediaPlayerElementSample2"
					    Source="ms-appx:///BasicApp/Assets/Videos/Getting_Started_with_Uno_Platform_for_Figma.mp4"
					   PosterSource="ms-appx:///BasicApp/Assets/Thumbnails/Getting_Started_with_Uno_Platform_for_Figma.png"
							AreTransportControlsEnabled="True"
							AutoPlay="False"
							MaxWidth="720" />

SCROLLABLE CONTENT
    </StackPanel>
</ScrollViewer>
```
Sample:
[BasicApp.zip](https://github.com/unoplatform/uno/files/12873838/BasicApp.zip)

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

_No response_

### NuGet package version(s)

```
<PackageReference Include="Uno.Resizetizer" Version="1.2.0-dev.68" />
<PackageReference Include="Uno.WinUI.Skia.Gtk" Version="5.0.0-dev.3431" />
<PackageReference Include="CommunityToolkit.Mvvm" Version="8.1.0" />
<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
<PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.6" />
<PackageReference Include="SkiaSharp.Skottie" Version="2.88.3" />
<PackageReference Include="Uno.WinUI.RemoteControl" Version="5.0.0-dev.3431" Condition="'$(Configuration)'=='Debug'" />
<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="5.0.0-dev.3431" />
<PackageReference Include="VideoLAN.LibVLC.Windows" Version="3.0.18" />
<PackageReference Include="Uno.WinUI.MediaPlayer.Skia.Gtk" Version="5.0.0-dev.3431" />
```

### Affected platforms

Skia (GTK on Linux/macOS/Windows)

### IDE

Visual Studio 2022

### IDE version

17.7.4

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #13954

## PR Type

What kind of change does this PR introduce?
- Bugfix

### <samp>🤖 Generated by Copilot at 3c7202f</samp>

Improved the video layout handling for the `GtkMediaPlayer` class and the `MediaPlayerPresenterExtension` class in the `Uno.UI.MediaPlayer.Skia.Gtk` add-in. This is to ensure that the video layout adapts to the size and orientation of the window and the content view.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by ramezgerges** [comment]
There is still a problem with the native mediaplayer not being clipped when partially scrolled off the screen. I've tried something similar to #13828 but I couldn't get things to work. resizing the window that contains the media player doesn't seem to do anything. Resizing the media player itself with `CropGeometry` also didn't work as intended.

File: src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/MediaPlayerPresenterExtension.cs [hunk 1]
```diff
@@ -39,6 +39,9 @@ public MediaPlayerPresenterExtension(object owner)
 		_owner = presenter;
 		_player = new GtkMediaPlayer(_owner);
 		var contentView = new ContentControl();
+
+		contentView.EffectiveViewportChanged += (_, _) => _player.UpdateVideoLayout();
```
Comment (kazo0) [inline src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/MediaPlayerPresenterExtension.cs:43]: Do we need to worry about unsubscribing from this event at any point?

Comment (ramezgerges) [reply src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/MediaPlayerPresenterExtension.cs:43]: I'm assuming the lifetime of the contentView matches that of the presenter itself. In that case, shouldn't we be ok? cc. @RafaelMendesRosa

**Comment by MartinZikmund** [comment]
Shouldn't this change be applied in a more general manner to native control embedding in `ContentPresenter` potentially?

**Comment by ramezgerges** [comment]
I've thought about that, but I'm not sure what a general solution would look like. It seems that every native element situation is somewhat different. For example, MPEs through through a lot of logic to not only position the media player, whereas something like textbox may be as simple as a `widget.setposition(managedX, managedY)`. You could have something like a delegate that each native wrapper provides for the custom details, but I guess the number of native elements doesn't warrant that complexity?

**Review by jeromelaban — APPROVED** [review]
We could have the change be more general, but this change does not seem likely to broken by a generalization.

**Comment by jeromelaban** [comment]
@ramezgerges if you think the changes are final, let's merge.

**Comment by ramezgerges** [comment]
Yes, the change seems simple (and logical) enough. Let's merge :).

**Pull request merged**